File: Programmers/고득점Kit/k번째수.py
def solution(array, commands):
    answer = []
    
    for command in commands:
        i = command[0] - 1
        j = command[1]
        k = command[2] - 1
        # print(array[i:j])
        new_array = sorted(array[i:j])
        # print(new_array[k])
        answer.append(new_array[k])

    return answer

File: Programmers/고득점Kit/k번째수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for (int idx = 0; idx < commands.length; idx++) {
            int i = commands[idx][0] - 1;
            int j = commands[idx][1];
            int k = commands[idx][2] - 1; 
            int[] newArr = Arrays.copyOfRange(array, i, j);
            //System.out.println(newArr);
            Arrays.sort(newArr);
            answer[idx] = newArr[k];
        }
        
        return answer;
    }
}

/*
import java.util.Arrays;
[배열 복사] : Arrays.copyOfRange(arr, startIndex, endIndex+1);
 */